client/js/components/back-to-top.js

```diff
@@ -0,0 +1,31 @@
+import {onWindowScrollThrottle$} from '../utils/dom-events';
+
+const toggleClass = (el, className, value) => {
+  el.classList.toggle(className, value);
+};
+
+const getVisibilityZoneStart = () => window.innerHeight;
+
+const getCurrentScrollPosition = () => window.pageYOffset;
+
+const getVisibilityZoneEnd = function(el) {
+  return Math.abs(document.body.getBoundingClientRect().top) + el.parentNode.getBoundingClientRect().bottom - getVisibilityZoneStart();
+};
+
+const determineVisibility = (visibilityZoneStart, visibilityZoneEnd, currentScrollPosition) => {
+  if (visibilityZoneStart < currentScrollPosition && currentScrollPosition < visibilityZoneEnd) {
```

Comment:
You could just `return visibilityZoneStart < currentScrollPosition && currentScrollPosition < visibilityZoneEnd;`